packages/devtools_app/lib/src/eval_on_dart_library.dart

```diff
@@ -272,3 +436,83 @@ class LibraryNotFound implements Exception {
 
   String get message => 'Library matchining one of $candidateNames not found';
 }
+
+Result<T> parseSentinel<T>(Object value) {
+  // TODO(rrousselGit) remove when NNBD lands
+  if (value == null) return Result.data(null);
+
+  if (value is T) {
+    return Result.data(value);
+  } else if (value is Sentinel) {
+    return Result.error(
+      SentinelException(value),
+      StackTrace.current,
+    );
+  } else {
+    return Result.error(
+      UnsupportedError('unkown type ${value.runtimeType}'),
+      StackTrace.current,
+    );
+  }
+}
+
+class UnknownEvalException implements Exception {
+  UnknownEvalException({
+    @required this.expression,
+    @required this.scope,
+    @required this.exception,
+  });
+
+  final String expression;
+  final Object exception;
+  final Map<String, String> scope;
+
+  @override
+  String toString() {
+    return 'Unknown error during the evaluation of `$expression`: $exception';
+  }
+}
+
+class SentinelException implements Exception {
+  SentinelException(this.sentinel);
+
+  final Sentinel sentinel;
+
+  @override
+  String toString() {
+    return 'SentinelException(sentinel: $sentinel)';
+  }
+}
+
+class EvalSentinelException extends SentinelException {
+  EvalSentinelException({
+    @required this.expression,
+    @required this.scope,
+    @required Sentinel sentinel,
+  }) : super(sentinel);
+
+  final String expression;
+  final Map<String, String> scope;
+
+  @override
+  String toString() {
+    return 'Evaluation `$expression` returned the Sentinel $sentinel';
+  }
+}
+
+class EvalErrorException implements Exception {
+  EvalErrorException({
+    @required this.expression,
+    @required this.scope,
+    @required this.errorRef,
+  });
+
+  final ErrorRef errorRef;
+  final String expression;
+  final Map<String, String> scope;
+
+  @override
+  String toString() {
+    return 'Evaluation `$expression` failed with $errorRef';
+  }
+}
```

Comment:
I would suggest keeping the changes to devtools core functionality in this CL and moving changes that are adding new pages to a separate CL. that will make the review easier.